nixos/modules/services/audio/mympd.nix

```
@@ -0,0 +1,94 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.services.mympd;
+in {
+  options = {
+
+    services.mympd = {
+
+      enable = lib.mkEnableOption (lib.mdDoc "MyMPD server");
+
+      package = lib.mkPackageOption pkgs "mympd" {};
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        description = lib.mdDoc ''
+          The HTTP port where mympd's web interface will be available.
+
+          The HTTPS/SSL port can be configured via {option}`config`.
+        '';
+        example = "8080";
+      };
+
+      ssl = lib.mkOption {
+        type = lib.types.bool;
+        description = lib.mdDoc ''
+          Whether to enable listening on the SSL port.
+
+          Refer to <https://jcorporation.github.io/myMPD/configuration/configuration-files#ssl-options>
+          for more information.
+        '';
+        default = false;
+      };
+
+      config = lib.mkOption {
+        type = with lib.types;
+          attrsOf
+            (nullOr
+              (oneOf [
+                str
+                bool
+                int
+              ]));
```

```suggestion
        type = with lib.types; attrsOf (nullOr (oneOf [ str bool int ]));
```
even if this is slightly less readable, we usually write it into one or a few lines